tests/cephadm/test_cephadm_upgrade.py

```diff
@@ -3,10 +3,88 @@
 
 """
 import logging
+from concurrent.futures import ThreadPoolExecutor
+from time import time
 
 from ceph.ceph_admin.orch import Orch
 
-LOG = logging.getLogger()
+LOG = logging.getLogger(__name__)
+
+
+class RadosBench:
+    def __init__(self, client, pool=""):
+        """
+        Initialize RadosBench
+        """
+        self.client = client
+        self.pool = pool or self.create_pool()
+
+    @staticmethod
+    def create_run_name():
+        """
+        Return run name
+        """
+        return f"test-run-{int(time())}"
+
+    def create_pool(self):
+        """
+        Create pool and enable application
+        """
+        name = f"pool-{int(time())}"
+        self.client.exec_command(cmd=f"ceph osd pool create {name} 128 128", sudo=True)
+        self.client.exec_command(
+            cmd=f"ceph osd pool application enable {name} rgw", sudo=True
+        )
+        return name
+
+    def delete_pool(self):
+        """
+        Delete pool
+        """
+        self.client.exec_command(
+            cmd="ceph config set global mon_allow_pool_delete true", sudo=True
+        )
+        self.client.exec_command(
+            cmd=f"ceph osd pool rm {self.pool} {self.pool} --yes-i-really-really-mean-it",
+            sudo=True,
+        )
```

Comment:
IMO,
- we could use a variable list holding the list of commands to be executed
- we could check if root level privileges is required.
- we could check if the configuration `mon_allow_pool_delete` is configured or not.
- we should fail if set to fails... so as to support negative testing.

Reply:
> In 5.x, we don't need to execute this with root level privileges... may we could add an attribute holding the RHCS version.

I Agree.

Reply:
> IMO,
> 
> * we could use a variable list holding the list of commands to be executed

Agreed.

> * we could check if root level privileges is required.

well its depends upon the client nodes used and RHCS version.

> * we could check if the configuration `mon_allow_pool_delete` is configured or not.

I think this should be handled in some library associated with pool.

> * we should fail if set to fails... so as to support negative testing.

yes, we could check this. But need to define these libraries.